RocketChat/Rocket.Chat
# [IMPROVE] Rewrite Two Factor Authentication Modal to React

## Proposed changes (including videos or screenshots)





## Issue(s)


## Steps to test or reproduce

- Go to Admin > Accounts > Two Factor Authentication and enable the following settings:
  - Enable Two Factor Authentication
  - Enable Two Factor Authentication via TOTP
  - Enable Two Factor Authentication via Email
  - Enforce Password fallback

- Now go to My Account > Security > Two Factor Authentication
- Click to enable Two factor via TOTP or Email (I use Authy on iOS for TOTP and configured a local email server to be able to receive the code for email)
- Once 2FA is enabled, try to disable it. The new modal for 2FA for the enabled method should pop up. You can also try to logout and login again, and the modal should be displayed.

- To test the password fallback, try to change and admin setting. It should require your password in order to save. (Beware that once you successfully type your password in once, it won't ask for it again for some time)
## Further comments
